sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/manager/DownloadManagerRequestTask.java

```diff
@@ -57,6 +57,24 @@ protected Void doInBackground(Void... params) {
              * IllegalArgumentException: Unknown URL content://downloads/my_download is thrown.
              */
             mDownloader.onDownloadError(new IllegalStateException("Failed to start download: Download Manager is disabled.", e));
+        } catch (Throwable throwable) {
+
+            /*
+             * We have to catch all types of errors to avoid crashes in runtime during downloading
+             * a new release with the specific Gradle configuration:
+             * release {
+             *     minifyEnabled true
+             *     proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+             * }
+             *
+             * Possible error:
+             * Caused by: java.lang.VerifyError: Verifier rejected class b.c.a.n.j.c.c: java.lang.Object
+             * b.c.a.n.j.c.c.doInBackground(java.lang.Object[]) failed to verify:
+             * java.lang.Object b.c.a.n.j.c.c.doInBackground(java.lang.Object[]): [0x6E]
+             * expected to be within a catch-all for an instruction where a monitor is held
+             * (declaration of 'b.c.a.n.j.c.c' appears in base.apk).
+             */
+            throw throwable;
```

Comment:
We just re-throwing it, could you please explain how exactly it fixes the crash?

Reply:
I explained it in the [comments](https://github.com/microsoft/appcenter-sdk-android/pull/1525/files#diff-26adf120c5910d7c6e54457267877a2e6637406f6a9d2f534367e86c5a13e224R74).